interfaces/builtin/desktop_session_control.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopSessionControlSummary = `allows setting up user processes for a desktop session`
+
+const desktopSessionBaseDeclarationSlots = `
+  desktop-session-control:
+    allow-installation:
+      slot-snap-type:
+        - app
```

Comment:
As written, I don't think it makes sense for an application snap to provide this as a plug. I think we only want core.

Reply:
Indeed, an oversight on my part. Fixed now.